daprdocs/content/en/concepts/components-concept.md

```diff
@@ -76,7 +88,20 @@ Lock components are used as a distributed lock to provide mutually exclusive acc
 
 ### Middleware
 
-Dapr allows custom [middleware]({{<ref "middleware.md">}})  to be plugged into the HTTP request processing pipeline. Middleware can perform additional actions on an HTTP request, such as authentication, encryption and message transformation before the request is routed to the user code, or before the response is returned to the client. The middleware components are used with the [service invocation]({{<ref "service-invocation-overview.md">}}) building block.
+Dapr allows custom [middleware]({{<ref "middleware.md">}}) to be plugged into the HTTP request processing pipeline. Middleware can perform additional actions on an HTTP request (such as authentication, encryption, and message transformation) before the request is routed to the user code, or the response is returned to the client. The middleware components are used with the [service invocation]({{<ref "service-invocation-overview.md">}}) building block.
 
 - [List of supported middleware components]({{< ref supported-middleware >}})
-- [Middleware implementations](https://github.com/dapr/components-contrib/tree/master/middleware)
\ No newline at end of file
+- [Middleware implementations](https://github.com/dapr/components-contrib/tree/master/middleware)
+
+## Pluggable components
+
+Dapr allows for users to create their own self-hosted components called pluggable components. These are components that do not need to be written in Go, exist outside the Dapr runtime and are able to "plug" into Dapr to utilize existing building block APIs.
+
+Where possible we encourage donating components to the Dapr project and community.
+
+However, pluggable components are used in scenarios where you want to create your own private component and choose not to include this into the Dapr project. This may be because you cannot include your component into the Dapr component repo since it is specific to your company or due to IP concerns. Or you want decouple your component updates from the Dapr release cycle.
+
+For more information read [Pluggable components overview]({{<ref "pluggable-components-overview">}})
+
+{{% alert title="Note" color="primary" %}} Since pluggable components are not required to be written in Go, they follow a different implementation process than built-in Dapr components. For more information on developing built-in components read [developing new components](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md)
```

Comment:
```suggestion
{{% alert title="Note" color="primary" %}} 
Since pluggable components are not required to be written in Go, they follow a different implementation process than built-in Dapr components. For more information on developing built-in components, read [developing new components](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md).
```